### PR TITLE
Prmp 1120 - NRL Lambda and SQS

### PR DESCRIPTION
**Pull request by jack-nhs**

### CODE REVIEW

File: infrastructure/variable.tf [hunk 1]
```diff
@@ -234,3 +234,7 @@ locals {
   current_region     = data.aws_region.current.name
   current_account_id = data.aws_caller_identity.current.account_id
 }
+
+variable "nrl_api_endpoint" {
+  default = "api.service.nhs.uk/record-locator/producer/FHIR/R4/DocumentReference"
```
Comment (chrisbloe-nhse) [inline infrastructure/variable.tf:239]: Always production?

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    APPCONFIG_ENVIRONMENT   = module.ndr-app-config.app_config_environment_id
+    APPCONFIG_CONFIGURATION = module.ndr-app-config.app_config_configuration_profile_id
+    WORKSPACE               = terraform.workspace
+    NRL_API_ENDPOINT        = local.is_production ? "https://${var.nrl_api_endpoint}" : "https://int.${var.nrl_api_endpoint}"
+    NRL_END_USER_ODS_CODE   = data.aws_ssm_parameter.end_user_ods_code.name
+  }
+  is_gateway_integration_needed = false
+  is_invoked_from_gateway       = false
+
+  depends_on = [
```
Comment (chrisbloe-nhse) [inline infrastructure/lambda-manage-nrl-pointer.tf:27]: This is implicit.

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+module "manage-nrl-pointer-alarm" {
+  source               = "./modules/lambda_alarms"
+  lambda_function_name = module.manage-nrl-pointer-lambda.function_name
+  lambda_timeout       = module.manage-nrl-pointer-lambda.timeout
+  lambda_name          = "manage_nrl_pointer_handler"
+  namespace            = "AWS/Lambda"
+  alarm_actions        = [module.manage-nrl-pointer-alarm-topic.arn]
+  ok_actions           = [module.manage-nrl-pointer-alarm-topic.arn]
+  depends_on           = [module.manage-nrl-pointer-lambda, module.manage-nrl-pointer-alarm-topic]
```
Comment (chrisbloe-nhse) [inline infrastructure/lambda-manage-nrl-pointer.tf:40]: These are implicit.

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          "ArnLike" : {
+            "aws:SourceArn" : "arn:aws:cloudwatch:eu-west-2:${data.aws_caller_identity.current.account_id}:alarm:*"
+          }
+        }
+        "Resource" : "*"
+      }
+    ]
+  })
+
+  depends_on = [module.manage-nrl-pointer-lambda, module.sns_encryption_key]
```
Comment (chrisbloe-nhse) [inline infrastructure/lambda-manage-nrl-pointer.tf:71]: implicit.

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    filter {
+      pattern = jsonencode({
+        body = {
+          "action" : [""]
+        }
+      })
+    }
+  }
+
+  depends_on = [
```
Comment (chrisbloe-nhse) [inline infrastructure/lambda-manage-nrl-pointer.tf:88]: Implicit

**Review by chrisbloe-nhse — DISMISSED** [review]
Thanks for making those changes :) All approved.

**Review by steph-torres-nhs — DISMISSED** [review]

File: infrastructure/lambda-manage-nrl-pointer.tf
Comment (robg-test) [inline infrastructure/lambda-manage-nrl-pointer.tf:1]: This lambda may need to be bumped up from the default memory setting due to it interacting with a 3rd Party API

**Review by AndyFlintNHS — APPROVED** [review]
Changes since last approval are a rebase from PRMP-1185 - will approve on this basis.

**Pull request merged**